libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/RtspMediaTrack.java

```diff
@@ -211,11 +212,18 @@ public int hashCode() {
         checkArgument(channelCount != C.INDEX_UNSET);
         checkArgument(!fmtpParameters.isEmpty());
         if(mediaEncoding.equals(RtpPayloadFormat.RTP_MEDIA_MPEG4_AUDIO)) {
-          Pair<Integer, Integer> cfgOpts = getSampleRateAndChannelCountFromAudioConfig(
-              fmtpParameters, channelCount, clockRate);
-          channelCount = cfgOpts.first;
-          clockRate = cfgOpts.second;
-          formatBuilder.setSampleRate(clockRate).setChannelCount(channelCount);
+          boolean isCPresent = true;
+          if (fmtpParameters.get(PARAMETER_MP4A_CPRESENT) != null && fmtpParameters.get(
+              PARAMETER_MP4A_CPRESENT).equals("0")) {
+            isCPresent = false;
+          }
+          @Nullable String configInput = fmtpParameters.get(PARAMETER_MP4V_CONFIG);
+          if (!isCPresent && configInput != null && configInput.length() % 2 == 0) {
+            Pair<Integer, Integer> configParameters = getSampleRateAndChannelCount(configInput);
+            channelCount = configParameters.first;
+            clockRate = configParameters.second;
+            formatBuilder.setSampleRate(clockRate).setChannelCount(channelCount);
```

Comment:
would it make sense to do `formatBuilder.setSampleRate(clockRate).setChannelCount(channelCount);` in `processAacFmtpAttributes` since the method receives `channelCount` and `clockRate` already?

Have to make sure this change works for LATM (c present or not) and ADTS though

Reply:
By default "formatBuilder.setSampleRate(clockRate).setChannelCount(channelCount);" is called at line 206. I'm calling this here only to update older values with value i'm getting after parsing.